packages/@aws-cdk/core/lib/stack-synthesizers/default-synthesizer.ts

```diff
@@ -362,6 +376,10 @@ export class DefaultStackSynthesizer extends StackSynthesizer {
     };
   }
 
+  public enrichContext(ctx: cxschema.MissingContext): cxschema.MissingContext {
+    return { ...ctx, lookupRoleArn: this.lookupRoleArn } ;
```

Comment:
The fact that we're doing this now also requires that the bootstrap stack is version 6

Reply:
I was planning to do this check in `synthesize()`. Something like:

```
const bootstrapVersion = session.assembly.hasMissingContext()
  ? MIN_BOOTSTRAP_STACK_VERSION_WITH_LOOKUP_ROLE
  : MIN_BOOTSTRAP_STACK_VERSION;

...

this.emitStackArtifact(this.stack, session, {
    assumeRoleArn: this._deployRoleArn,
    cloudFormationExecutionRoleArn: this._cloudFormationExecutionRoleArn,
    stackTemplateAssetObjectUrl: templateManifestUrl,

    requiresBootstrapStackVersion: bootstrapVersion, // <-- It was a constant before

    bootstrapStackVersionSsmParameter: `/cdk-bootstrap/${this.qualifier}/version`,
    additionalDependencies: [artifactId],
});

```




So we would have only one place where this check is performed. What do you think?

Reply:
Discussed offline.